### PR TITLE
Google crawler refactoring

### PR DESCRIPTION
**Pull request by Oplia512**
refactored and added few tests

### CODE REVIEW

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/processor/core/CrawlerProcessorListener.java [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-import java.util.List;
+import java.util.function.Supplier;
 
 /**
  * @author Siarhei Shauchenka
  * @since 17.08.17
  */
 public interface CrawlerProcessorListener {
-    void success(List<BookDiscountData> bookData, Paginator paginator);
+    void success(Supplier<ResultModel> resultSupplier);
```
Comment (mprtcz) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/processor/core/CrawlerProcessorListener.java:13]: :+1:

File: googlecrawler/src/test/java/com/blackbook/googlecrawler/impl/GoogleCrawlerTest.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.blackbook.googlecrawler.impl;
+
+import org.testng.annotations.Test;
+
+/**
+ * @author Siarhei Shauchenka at 07.09.17
+ */
+@Test
+public class GoogleCrawlerTest {
+
```
Comment (mprtcz) [inline googlecrawler/src/test/java/com/blackbook/googlecrawler/impl/GoogleCrawlerTest.java]: Dat test :suspect:

**Review by mprtcz — APPROVED** [review]

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/processor/core/CrawlerProcessorListener.java (hunk 1 above)
Comment (agawda) [reply googlecrawler/src/main/java/com/blackbook/googlecrawler/processor/core/CrawlerProcessorListener.java:13]: 👍

File: googlecrawler/src/test/java/com/blackbook/googlecrawler/impl/GoogleCrawlerTest.java (hunk 2 above)
Comment (agawda) [reply googlecrawler/src/test/java/com/blackbook/googlecrawler/impl/GoogleCrawlerTest.java]: 🥇

File: googlecrawler/src/test/java/com/blackbook/googlecrawler/parser/GoogleParserTest.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        Assert.assertEquals(discountData.getBookDiscountDetails(), TEST_DISCOUNT_RESULT);
+
+        BookData bookData = discountData.getBookData();
+
+        Assert.assertEquals(bookData.getTitle(), TEST_TITLE);
+        Assert.assertEquals(bookData.getSubtitle(), TEST_SUBTITLE);
+        Assert.assertEquals(bookData.getAuthors(), TEST_AUTHORS_RESULT);
+        Assert.assertEquals(bookData.getGenre(), TEST_CATEGORIES_RESULT);
+        Assert.assertEquals(bookData.getCoverUrl(), TEST_IMAGE_URL);
+        Assert.assertEquals(bookData.getBookPageUrl(), TEST_BOOK_PAGE_URL);
```
Comment (agawda) [inline googlecrawler/src/test/java/com/blackbook/googlecrawler/parser/GoogleParserTest.java:158]: You should use soft assertions if you want to check so many things in one test.

**Review by agawda — APPROVED** [review]

**Pull request merged**